detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeEnvironmentParser.java

```diff
@@ -0,0 +1,48 @@
+package com.synopsys.integration.detectable.detectables.bitbake.parse;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detectable.detectables.bitbake.model.BitbakeEnvironment;
+
+public class BitbakeEnvironmentParser {
+    private static final String ARCHITECTURE_VARIABLE_NAME = "MACHINE_ARCH";
+    private static final String LICENSES_DIR_VARIABLE_NAME = "LICENSE_DIRECTORY";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final Predicate<String> isArchitectureLine =  l -> l.startsWith(ARCHITECTURE_VARIABLE_NAME+"=");
+    private final Predicate<String> isLicensesDirLine =  l -> l.startsWith(LICENSES_DIR_VARIABLE_NAME +"=");
+
+    public BitbakeEnvironment parseArchitecture(List<String> bitbakeEnvironmentCmdOutput) {
+        Optional<String> architecture = bitbakeEnvironmentCmdOutput.stream()
+            .filter(isArchitectureLine)
+            .map(line -> isolateVariableValue(line, ARCHITECTURE_VARIABLE_NAME))
+            .map(this::unquote)
+            .findFirst();
+
+        Optional<String> licensesDirPath = bitbakeEnvironmentCmdOutput.stream()
```

Comment:
You're able to get the license directory from the environment output? That's handy!